consensus/types/src/light_client_header.rs

```diff
@@ -1,26 +1,223 @@
+            ),
+            ForkName::Deneb => LightClientHeader::Deneb(
+                LightClientHeaderDeneb::block_to_light_client_header(block)?,
+            ),
+        };
+        Ok(header)
+    }
+
+    pub fn from_ssz_bytes(bytes: &[u8], fork_name: ForkName) -> Result<Self, ssz::DecodeError> {
+        let header = match fork_name {
+            ForkName::Altair | ForkName::Merge => {
+                let header = LightClientHeaderAltair::from_ssz_bytes(bytes)?;
+                LightClientHeader::Altair(header)
+            }
+            ForkName::Capella => {
+                let header = LightClientHeaderCapella::from_ssz_bytes(bytes)?;
+                LightClientHeader::Capella(header)
+            }
+            ForkName::Deneb => {
+                let header = LightClientHeaderDeneb::from_ssz_bytes(bytes)?;
+                LightClientHeader::Deneb(header)
+            }
+            ForkName::Base => {
+                return Err(ssz::DecodeError::BytesInvalid(format!(
+                    "LightClientHeader decoding for {fork_name} not implemented"
+                )))
+            }
+        };
+
+        Ok(header)
+    }
+}
+
+impl<E: EthSpec> Encode for LightClientHeader<E> {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn ssz_bytes_len(&self) -> usize {
+        match self {
+            LightClientHeader::Altair(header) => header.ssz_bytes_len(),
+            LightClientHeader::Capella(header) => header.ssz_bytes_len(),
+            LightClientHeader::Deneb(header) => header.ssz_bytes_len(),
+        }
+    }
+
+    fn ssz_fixed_len() -> usize {
+        ssz::BYTES_PER_LENGTH_OFFSET
+    }
+
+    fn as_ssz_bytes(&self) -> Vec<u8> {
+        match self {
+            LightClientHeader::Altair(header) => header.as_ssz_bytes(),
+            LightClientHeader::Capella(header) => header.as_ssz_bytes(),
+            LightClientHeader::Deneb(header) => header.as_ssz_bytes(),
+        }
+    }
+
+    fn ssz_append(&self, buf: &mut Vec<u8>) {
+        match self {
+            LightClientHeader::Altair(header) => header.ssz_append(buf),
+            LightClientHeader::Capella(header) => header.ssz_append(buf),
+            LightClientHeader::Deneb(header) => header.ssz_append(buf),
+        }
+    }
+}
+
+impl<E: EthSpec> LightClientHeaderAltair<E> {
+    pub fn block_to_light_client_header(block: &SignedBeaconBlock<E>) -> Result<Self, Error> {
+        Ok(LightClientHeaderAltair {
+            beacon: block.message().block_header(),
+            phantom_data: PhantomData,
+        })
+    }
+}
+
+impl<E: EthSpec> LightClientHeaderCapella<E> {
+    pub fn block_to_light_client_header(block: &SignedBeaconBlock<E>) -> Result<Self, Error> {
+        let payload = block
+            .message()
+            .execution_payload()?
+            .execution_payload_capella()?
+            .to_owned();
+
+        let header = ExecutionPayloadHeaderCapella::from(&payload);
```

Comment:
We can remove this clone here:

```suggestion
            .execution_payload_capella()?;

        let header = ExecutionPayloadHeaderCapella::from(payload);
```